web/core/store/inbox/project-inbox.store.ts

```diff
@@ -329,20 +329,23 @@ export class ProjectInboxStore implements IProjectInboxStore {
     }
   };
 
-  toggleIntakeForms = async (workspaceSlug: string, projectId: string, isDisabled: boolean) => {
+  toggleIntakeForms = async (workspaceSlug: string, projectId: string, data: Partial<TInboxForm>) => {
+    const initialData = this.intakeForms[projectId];
     try {
       runInAction(() => {
-        set(this.intakeForms, projectId, { ...this.intakeForms[projectId], is_disabled: isDisabled });
+        set(this.intakeForms, projectId, { ...this.intakeForms[projectId], ...data });
+      });
+      const result = await this.inboxIssueService.updatePublishForm(workspaceSlug, projectId, data);
+      runInAction(() => {
+        set(this.intakeForms, projectId, { ...this.intakeForms[projectId], anchor: result?.anchor });
```

Comment:
You are setting data in line 336, and not setting it in 340?

Reply:
Data is already being set optimistically, anchor is what comes from the api response so setting it after the api.